flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -554,6 +554,16 @@ public class ExecutionConfigOptions {
                                             + "all changes to downstream just like when the mini-batch is "
                                             + "not enabled.");
 
+    @Documentation.TableOption(execMode = Documentation.ExecMode.STREAMING)
+    public static final ConfigOption<Integer> UNBOUNDED_OVER_VERSION =
+            ConfigOptions.key("table.exec.unbounded-over.version")
+                    .intType()
+                    .defaultValue(1)
+                    .withDescription(
+                            "Which version of the unbounded over aggregation to use: "
+                                    + " 1 - legacy version"
+                                    + " 2 - new version with improved performance");
```

Comment:
nit: remove the word new so it does not need to be changed if we add a 3rd value at some stage

Reply:
If this goes into the first version 2 - should we default to 2?

Reply:
I think we can default to version 2 indeed and keep this as an opt-out feature.